### ISSUE TITLE
Online FW upgrade not working?

### ISSUE THREAD

**Issue by vin-w** [roleoroleo/yi-hack-MStar#526]
Recently I found all camera does not show available version for new firmware.

"Online FW upgrade 
Installed: 0.5.3 - Available:"

**Comment by roleoroleo** [comment]
Did you filter the http/https connection?

**Comment by vin-w** [comment]
I remember I once enable proxychain but disable afterwards.

Here is some output I checked.

```
/home/yi-hack/bin # wget -O -  https://api.github.com/repos/roleoroleo/yi-hack-MStar/releases/latest
Connecting to api.github.com (20.205.243.168:443)
wget: note: TLS certificate validation not implemented
wget: TLS error from peer (alert code 80): 80
wget: error getting response: Connection reset by peer

/home/yi-hack/bin # cat /home/yi-hack/etc/proxychains.conf

```

**Comment by roleoroleo** [comment]
proxychains is not a problem if you don't use the Yi platform.
The log `wget: note: TLS certificate validation not implemented` is normal.

This is not ok:
```
wget: TLS error from peer (alert code 80): 80
wget: error getting response: Connection reset by peer
```

Check your connection, firewall/proxy, router.
It should work.
Try with a temporary tethering.

**Comment by marchidaniele** [comment]
I have same problem.
I have a cam y25 with base version 4.2.0.0H_201909041620 and Firmware Version 0.4.9.

Online update don't work 
<img width="604" alt="Schermata 2023-12-20 alle 00 54 55" src="https://github.com/roleoroleo/yi-hack-MStar/assets/129943075/e0e7d0b6-647e-4cdb-939d-532c0c0c1477">

and if I connect via ssh and download latest firmware

```
/tmp/sd # wget -O -  https://api.github.com/repos/roleoroleo/yi-hack-MStar/releases/latest
Connecting to api.github.com (140.82.121.6:443)
wget: note: TLS certificate validation not implemented
wget: TLS error from peer (alert code 80): 80
wget: error getting response: Connection reset by peer
```
if you download another file similar error:
```
/tmp/sd # wget https://ubuntu.mirror.garr.it/releases/jammy/ubuntu-22.04.3-live-server-amd64.iso
Connecting to ubuntu.mirror.garr.it (90.147.160.70:443)
wget: note: TLS certificate validation not implemented
wget: TLS error from peer (alert code 40): handshake failure
wget: error getting response: Connection reset by peer
```
and if I add switch --no-check-certificates don't help:
```
/tmp/sd # wget -O -  https://api.github.com/repos/roleoroleo/yi-hack-MStar/releases/latest --no-check-certificates
wget: unrecognized option '--no-check-certificates'
```

any chance for upgrade my cam?

**Comment by roleoroleo** [comment]
I need to update busybox version.
Meanwhile you can upgrade your cam with this procedure:
https://github.com/roleoroleo/yi-hack-MStar/wiki/Manual-firmware-upgrade

**Comment by roleoroleo** [comment]
It's a bigger problem than it initially seemed.
Currently I'm unable to fix it.

**Comment by roleoroleo** [comment]
Fixed here:
https://github.com/roleoroleo/yi-hack-MStar/commit/e86b722f85cb042ca630e2b805e462eecc9db0b0

**Comment by Traoma** [comment]
Hello, What's the way to upgrade to the last version with very old release? 
I'm with the 0.3.5 that has not yet the local firmware update implementation (came since 0.4.8)

**Comment by roleoroleo** [comment]
Save the config.
Copy home_xxx and sys_xxx to your sd card.
Reboot.
Restore the config if you lost it.

**Comment by roleoroleo** [comment]
Are you sure that you are using the correct files for your model?

**Comment by Traoma** [comment]
It's OK. The problem was the sd card ... I put on old (but working) 4gb card. Tried with a new 32gb and the update has worked. Thanks

**Comment by Madaraxa** [comment]
Online Update from 0.5.4 to 0.5.5 wont work on my two Mstar cams.
(my two allwinner worked from 0.3.8 to 0.3.9)

**Comment by Extreme79** [comment]
Happened to me today, from 0.53, in 3 cam. 
Solved upgrading manually (but I lost settings).
1 identical camera, 0.53, upgraded flawlessy without loosing config.

**Comment by roleoroleo** [comment]
The problem was fixed with 0.5.4.
If you come from 0.5.3 it doesn't work.

> Online Update from 0.5.4 to 0.5.5 wont work on my two Mstar cams. I guess this is fixed from 0.5.5 and above? So this is for now the last manual Update needed? (my two allwinner worked from 0.3.8 to 0.3.9)

Reading better your message, this is not correct. The update should have worked.
Let me know if you have problems with the next future update.

**Comment by Madaraxa** [comment]
Hi @roleoroleo . I just tried to online Update from 0.5.5 to 0.5.6 and it didnt work 4 Times. I came to github to See if there is a problem and found the rtsp stream Not working issue which shall be fixed in 0.5.7. I stopped trying to Update now because of the upcoming 0.5.7.
But shouldnt the Update work from 0.5.5 to 0.5.6? Or did you delete 0.5.6?
